gapic/schema/wrappers.py

```diff
@@ -757,17 +757,80 @@ class HttpRule:
     uri: str
     body: Optional[str]
 
-    @property
-    def path_fields(self) -> List[Tuple[str, str]]:
+    def path_fields(self, method: "~.Method") -> List[Tuple[Field, str, str]]:
         """return list of (name, template) tuples extracted from uri."""
-        return [(match.group("name"), match.group("template"))
+        input = method.input
+        return [(input.get_field(*match.group("name").split(".")), match.group("name"), match.group("template"))
                 for match in path_template._VARIABLE_RE.finditer(self.uri)]
 
-    @property
-    def sample_request(self) -> str:
+    def sample_request(self, method: "~.Method") -> str:
         """return json dict for sample request matching the uri template."""
-        sample = utils.sample_from_path_fields(self.path_fields)
-        return json.dumps(sample)
+
+        def sample_from_path_fields(paths: List[Tuple["wrappers.Field", str, str]]) -> Dict[Any, Any]:
+            """Construct a dict for a sample request object from a list of fields
+               and template patterns.
+
+            Args:
+                  paths: a list of tuples, each with a (segmented) name and a pattern.
+            Returns:
+                  A new nested dict with the templates instantiated.
+            """
+
+            request: Dict[str, Any] = {}
+
+            def _sample_names() -> Generator[str, None, None]:
```

Comment:
Iterator[str] instead?

[How to annotate a generator in python3?](https://stackoverflow.com/a/63237329)

Also, could we do something like naming the samples based on the last component of the path? It might make the sample names look a little better.

Reply:
I would just a soon rip out all the type annotation. `Iterator[str]` does look more correct, will fix